### PR TITLE
Replace FindFirstFileExW with FindFirstFileW

### LINKED ISSUE #16804: Remove-Item fails on FAT32 drive when encountering some Unicode characters
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Consider the data in
[RADWIMPS.zip](https://github.com/PowerShell/PowerShell/files/7941251/RADWIMPS.zip)
(note: this is copyrighted material I rip-ped from a CD I legally own; please don't redistribute).

When this is found on a FAT32 drive, you cannot delete the file tree using `Remove-Item`:

 
image

This repro's both in 7.2.1 and 7.3.0-preview.1.

CMD.EXE has no problem with the same data:

image

### Expected behavior

```console
`Remove-Item` should remove the entire directory tree.
```

### Actual behavior

```console
`Remove-Item` fails with `Remove-Item: No mapping for the Unicode character exists in the target multi-byte code page.`
```

### Error details

```console
Exception             :
    Type            : System.ComponentModel.Win32Exception
    NativeErrorCode : 1113
    ErrorCode       : -2147467259
    TargetSite      :
        Name          : IsReparsePointLikeSymlink
        DeclaringType : Microsoft.PowerShell.Commands.InternalSymbolicLinkLinkCodeMethods
        MemberType    : Method
        Module        : System.Management.Automation.dll
    Message         : No mapping for the Unicode character exists in the target multi-byte code page.
    Source          : System.Management.Automation
    HResult         : -2147467259
    StackTrace      :
   at
Microsoft.PowerShell.Commands.InternalSymbolicLinkLinkCodeMethods.IsReparsePointLikeSymlink(FileSystemInfo
fileInfo)
   at Microsoft.PowerShell.Commands.FileSystemProvider.RemoveDirectoryInfoItem(DirectoryInfo directory,
Boolean recurse, Boolean force, Boolean rootOfRemoval)
   at Microsoft.PowerShell.Commands.FileSystemProvider.RemoveDirectoryInfoItem(DirectoryInfo directory,
Boolean recurse, Boolean force, Boolean rootOfRemoval)
   at Microsoft.PowerShell.Commands.FileSystemProvider.RemoveItem(String path, Boolean recurse)
   at System.Management.Automation.SessionStateInternal.RemoveItem(CmdletProvider providerInstance, String
path, Boolean recurse, CmdletProviderContext context)
CategoryInfo          : NotSpecified: (:) [Remove-Item], Win32Exception
FullyQualifiedErrorId : System.ComponentModel.Win32Exception,Microsoft.PowerShell.Commands.RemoveItemCommand
InvocationInfo        :
    MyCommand        : Remove-Item
    ScriptLineNumber : 1
    OffsetInLine     : 1
    HistoryId        : 78
    Line             : Remove-Item -Recurse -Force -LiteralPath J:\RADWIMPS -Verbose
    PositionMessage  : At line:1 char:1
                       + Remove-Item -Recurse -Force -LiteralPath J:\RADWIMPS -Verbose
                       + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    InvocationName   : Remove-Item
    CommandOrigin    : Internal
ScriptStackTrace      : at <ScriptBlock>, <No file>: line 1
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.3.0-preview.1
PSEdition                      Core
GitCommitId                    7.3.0-preview.1
OS                             Microsoft Windows 10.0.19044
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visuals

_No response_

### PR DESCRIPTION
**Pull request by iSazonov**
# PR Summary

Fix #16804

Test would be very slow so I added protection comment in the code since no functional changes was done.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Review by SteveL-MSFT — APPROVED** [review]

File: src/System.Management.Automation/namespaces/FileSystemProvider.cs [hunk 1]
```diff
@@ -8278,7 +8280,7 @@ internal static bool IsReparsePointLikeSymlink(FileSystemInfo fileInfo)
                 fullPath = PathUtils.EnsureExtendedPrefix(fullPath);
             }
 
-            using (SafeFindHandle handle = FindFirstFileEx(fullPath, FINDEX_INFO_LEVELS.FindExInfoBasic, ref data, FINDEX_SEARCH_OPS.FindExSearchNameMatch, IntPtr.Zero, 0))
```
Comment (daxian-dbw) [inline src/System.Management.Automation/namespaces/FileSystemProvider.cs:8281]: Can the structs `FINDEX_INFO_LEVELS` and `FINDEX_SEARCH_OPS` be removed now?

Comment (daxian-dbw) [reply src/System.Management.Automation/namespaces/FileSystemProvider.cs:8281]: It's better to remove all if they are not in use anymore. If for some reason we need to revert this change, all will be reverted back anyways.

Comment (iSazonov) [reply src/System.Management.Automation/namespaces/FileSystemProvider.cs:8281]: I am exploring now how we can make our providers better in particular File Provider and may need it again. I don't mind removing it for now though.

**Review by daxian-dbw — APPROVED** [review]

**Comment by sba923** [comment]
Great job!

Shall I correctly infer that this will surface in the next `7.2.x` and `7.3.0-preview.x`?

**Comment by iSazonov** [comment]
@sba923 You can subscribe to .Net issue https://github.com/dotnet/runtime/issues/64789 to track news about the API issue.

The fix will be in next PowerShell 7.3 Preview2. As for backport to LTS version, I guess MSFT team will weight.  Personally I don't see needs to backport the fix since it is rare scenario.

**Comment by TravisEz13** [comment]
@PowerShell/powershell-maintainers 
We only take security fixes and regression to LTS.  This is appropriate for 7.3.  Rejecting back-port.

**Pull request merged**